source/features/expand-all-collapsed-code.tsx

```diff
@@ -0,0 +1,77 @@
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import features from '../libs/features';
+
+// When diff table's childlist changes, select and click expand button to load more lines
+const clickOnAutoexpandObserver = new MutationObserver(mutations => {
+	for (const mutation of mutations) {
+		if (mutation.type === 'childList') {
+			const btn = select('.js-expand.directional-expander.single-expander', mutation.target as HTMLElement);
+			if (btn) {
+				btn.click();
+			}
+		}
+	}
+});
+
+function clickOnAutoexpand(element: HTMLElement): void {
+	clickOnAutoexpandObserver.observe(element, {childList: true});
+}
+
+function unfoldOnAltClick(event: DelegateEvent<MouseEvent>): void {
+	if (event.altKey) {
+		event.preventDefault();
+		unfold();
+	}
+}
+
+const addListenersObserver = new MutationObserver(mutations => {
+	for (const mutation of mutations) {
+		if (mutation.type === 'childList') {
+			delegate(
+				mutation.target as HTMLElement,
+				'.js-expand.directional-expander.single-expander',
+				'click',
+				unfoldOnAltClick
+			);
+		}
+	}
+});
+
+// When user clicks on expand button without alt, we need to add alt-click listeners to newly created buttons
+function addListenersOnExpand(element: HTMLElement): void {
+	addListenersObserver.observe(element, {childList: true});
+}
+
+function unfold(): void {
```

Comment:
If you write this, you can avoid `unfoldOnAltClick` altogether.

```suggestion
function unfold(event: DelegateEvent<MouseEvent>): void {
	if (!event.altKey) {
		return;
	}
	event.preventDefault();
```